lib/business/calendar.rb

```diff
@@ -19,13 +19,13 @@ def self.load(calendar)
       raise "No such calendar '#{calendar}'" unless directory
 
       yaml = YAML.load_file(File.join(directory, "#{calendar}.yml"))
-      valid_keys = ['holidays', 'working_days']
+      valid_keys = %w(holidays working_days working_dates)
 
       unless (yaml.keys - valid_keys).empty?
         raise "Only valid keys are: #{valid_keys.join(', ')}"
       end
-      
-      self.new(holidays: yaml['holidays'], working_days: yaml['working_days'])
+
+      self.new(valid_keys.map{|k| [k.to_sym, yaml[k]]}.to_h)
```

Comment:
I'd be in favour of keeping this explicit - it makes the code a bit more self-explanatory.